targets/go/make.js

```diff
@@ -149,6 +149,8 @@ function getModelPropertyDef(property, datatype) {
         return capitalizeFirstLetter(property.name) + " []" + basicType + " `json:\""+property.name+",omitempty\"`";
     else if (property.collection && property.collection === "map")
         return capitalizeFirstLetter(property.name) + " map[string]" + basicType + " `json:\""+property.name+",omitempty\"`";
+    else if (property.isclass && property.optional)
```

Comment:
nice catch!
do we need the property to be optional in order to make it a pointer?

Reply:
also, I did a similar change recently, you can see it here: https://github.com/PlayFab/SDKGenerator/compare/master...dgkanatsios:golang#diff-3dc480651ae9a37e182ff68e52f5badbR155-R158

however, your change (maybe without the `optional` clause) is better, maybe you can delete my change in order to commit yours :) thanks again!

Reply:
I may have made the optional requirement to avoid propagating a breaking change any further than necessary.

However, if the property is _not_ optional and is a class, then what benefit is there to making it a pointer? That introduces the possibility that it could be set to nil which wouldn't be valid?

Reply:
the reason I converted it to pointer now is that the test here was failing
https://github.com/dgkanatsios/playfabsdk-go/blob/master/e2e/main.go#L212

apparently, having the authentication.GetEntityTokenRequestModel{} have an empty Entity field (and not a nil) makes the EntityToken call fail. That's why I made it a pointer. Feel free to try it out and let me know!

Reply:
ping @Sumo-MBryant